src/PAModel/Utility/FilePath.cs

```diff
@@ -30,6 +33,29 @@ public string ToPlatformPath()
             return Path.Combine(_pathSegments.Select(Utilities.EscapeFilename).ToArray());
         }
 
+        /// <summary>
+        /// Checks if the path length is more than MAX_PATH (260)
+        /// then it truncates the path, and adds a hash to the filename while retaining the file extension.
+        /// </summary>
+        /// <param name="path">Full path of the file.</param>
+        /// <returns></returns>
+        public static string ToValidPath(string path)
+        {
+            if (path.Length > MAX_PATH)
+            {
+                var extension = path.EndsWith(yamlExtension)
```

Comment:
>path.EndsWith(yamlExtension) [](http://example.com/codeflow?start=32&length=28)

Can we use Path.GetExtension() ?

Reply:
the extension we use is .fx.yaml, the GetExtension() method only return .yaml, so one option would be assume that .yaml is always .fx.yaml but then if ever we have a file with just .yaml extension that might break our tool.